adapter/config/memory/monitor.go

```diff
@@ -0,0 +1,107 @@
+package memory
+
+import (
+	"reflect"
+	"sort"
+	"time"
+
+	"istio.io/pilot/model"
+)
+
+const ()
+
+type Configs []model.Config
+
+// Handler specifies a function to apply on a Config for a given event type
+type Handler func(model.Config, model.Event)
+
+type Monitor interface {
```

Comment:
I'm thinking of adding this in-memory controller into master. It seems incomplete as it doesn't properly compare poll result with existing cache. Are you interested in separating this controller out and finishing it, or this is purely for prototyping?